### DISCUSSION TITLE
What's the best way to get the AnyIO version number with Python?

### DISCUSSION THREAD

**Discussion by br3ndonland** [agronholm/anyio#713] [Q&A] [answered]
I've been working with AnyIO 3 for a while now and would like to add support for AnyIO 4. As shown in the [version history](https://anyio.readthedocs.io/en/stable/versionhistory.html), AnyIO 4 makes some backwards incompatible changes and adds some new features.

To help support multiple versions of a package, I would typically import and read the version number with something like `anyio.__version__`. I don't see a version object though.

The `pyproject.toml` shows that the version number is dynamic.

https://github.com/agronholm/anyio/blob/4a23745badf5bf5ef7928f1e346e9986bd696d82/pyproject.toml#L35

The only place I see the version number referenced is in the source code for the docs, where `importlib.metadata` is used.

https://github.com/agronholm/anyio/blob/4a23745badf5bf5ef7928f1e346e9986bd696d82/docs/conf.py#L4

https://github.com/agronholm/anyio/blob/4a23745badf5bf5ef7928f1e346e9986bd696d82/docs/conf.py#L21-L23

So is `importlib.metadata` the best way to get the installed version of AnyIO?

**Comment by agronholm** [comment, marked as answer]
Yes, that is the best way to get the version number of any installed Python library.
